### ISSUE TITLE
Class Property without initializer parsed when `classProperties` plugin is not enabled

### ISSUE THREAD

**Issue by DrewML** [babel/babylon#242]
### Input Code

```js
class Bar {
   abc; 
}
```

### Babylon/Babel Configuration

Calling `babylon.parse()` _without_ a `plugins` []

### Expected Behavior
Unexpected token error

### Current Behavior
Class Property is parsed

### Your Environment

| software         | version
| ---------------- | -------
| Babylon          |  6.14.1
| node             | All
| npm              | All
| Operating System | All

----

The only check for the `classProperties` plugin is currently in `pp.parseClassProperty` in `statement.js`:

```js
pp.parseClassProperty = function (node) {
  if (this.match(tt.eq)) {
    if (!this.hasPlugin("classProperties")) this.unexpected();
    this.next();
    node.value = this.parseMaybeAssign();
  } else {
    node.value = null;
  }
  this.semicolon();
  return this.finishNode(node, "ClassProperty");
};
```

If an initializer is not provided, it drops passed the branch that has the plugin check. I think we can just fix this by moving the plugin check to the top of `parseClassProperty`, since we've already done a check prior (in `pp.parseClassBody`) to verify we're dealing with a class prop.

**Comment by loganfsmyth** [comment]
This does indeed seem like a bug. Just a general note, any fix for this needs to take https://github.com/babel/babylon/blob/48873f886aa91c4d250ccce0c5e220c82f538f11/src/plugins/flow.js#L1050 into account. If I had to guess, that's the motivation for the current behavior.

You can do

```
class Foo {
    prop: string;
}
```

with only `flow` turned on, without `classProperties`.

**Comment by DrewML** [comment]
Good call on the `flow` stuff, hadn't considered that. Will keep in mind. I plan on addressing this, just trying to get through work on the `Private Class fields` proposal first.

**Comment by bakkot** [comment]
I believe this was fixed by #300.
